kong/clustering/rpc/manager.lua

```diff
@@ -42,15 +41,17 @@ function _M.new(conf, node_id)
     -- clients[node_id]: { socket1 => true, socket2 => true, ... }
     clients = {},
     client_capabilities = {},
-    client_ips = {},  -- store DP node's ip addr
     node_id = node_id,
     conf = conf,
     cluster_cert = assert(clustering_tls.get_cluster_cert(conf)),
     cluster_cert_key = assert(clustering_tls.get_cluster_cert_key(conf)),
     callbacks = callbacks.new(),
   }
 
-  self.concentrator = concentrator.new(self, kong.db)
+  if conf.role == "control_plane" then
+    self.concentrator = require("kong.clustering.rpc.concentrator").new(self, kong.db)
+    self.client_ips = {}  -- store DP node's ip addr
```

Comment:
What about make `client_ips` a part of concerntrator

Reply:
Not sure about it, but I think collecting ip addresses is not the responsibility of concentrator.

Reply:
Correct, `client_ips` is mainly for the directly connected node, concentrator uses database to fetch the list instead of a Lua table.